### PR TITLE
My Site Dashboard: add error messages when posts fails to load

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #17697

Add two types of error messages:

1. When trying to load drafts/scheduled and there are zero posts (it shouldn't happen but it's handled anyway)
2. When posts fail to load at all

### To test

Do a clean install, login into your account, and enable the flag (don't go to the Dashboard), then:

#### Failing to load

1. Close the app and turn the wi-fi off
2. Open the app and go to the Dashboard
3. You should see a message that we're unable to load posts right now
4. Turn on the wi-fi
5. Tap to retry
6. The posts should load and be displayed correctly

#### No draft posts

1. Pick a blog that doesn't have draft posts (or delete all that you have)
2. Open the dashboard
3. You should see a message saying that you don't have any posts
4. There's no option to retry

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        errorTitle.textAlignment = .center
+        errorTitle.textColor = .textSubtle
+        WPStyleGuide.configureLabel(errorTitle, textStyle: .callout, fontWeight: .semibold)
+        return errorTitle
+    }()
+
+    private lazy var retryLabel: UILabel = {
+        let retryLabel = UILabel()
+        retryLabel.textAlignment = .center
+        retryLabel.text = "Tap to retry"
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift]:
```suggestion
        retryLabel.text = NSLocalizedString("Tap to retry", comment: "...")
```

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        retryLabel.textColor = .textSubtle
+        WPStyleGuide.configureLabel(retryLabel, textStyle: .callout, fontWeight: .regular)
+        return retryLabel
+    }()
+
+    convenience init(message: String, retry: Bool) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift]:
```suggestion
    convenience init(message: String, canRetry: Bool) {
```

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewController.swift [hunk 3]
```diff
@@ -105,8 +117,21 @@ extension PostsCardViewController: PostsCardView {
     }
 
     func hideLoading() {
+        errorView?.removeFromSuperview()
         removeGhostableTableView()
     }
+
+    func showError(message: String, retry: Bool) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewController.swift:124]:
```suggestion
    func showError(message: String, canRetry: Bool) {
```

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewModel.swift [hunk 4]
```diff
@@ -121,17 +131,36 @@ private extension PostsCardViewModel {
...
     }
 
     enum Constants {
         static let numberOfPosts = 3
         static let numberOfPostsToSync: NSNumber = 4
     }
+
+    enum Strings {
+        static let noPostsMessage = NSLocalizedString("You don't have any posts", comment: "Displayed when the user views the dashboard posts card but they have no posts")
+        static let loadingFailure: String = NSLocalizedString("Unable to load posts right now.", comment: "Message for when posts fail to load on the dashboard")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewModel.swift]: [nit]
```suggestion
        static let loadingFailure = NSLocalizedString("Unable to load posts right now.", comment: "Message for when posts fail to load on the dashboard")
```

**Review by momo-ozawa — APPROVED** [review]
Works as expected! Left a couple of nitpicks / comments.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift]: Done in 7381befef4d2fe5b25b91e49eb5429c8b30904fd

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/Helpers/DashboardCardInnerErrorView.swift]: Done in e5177d4705ee7bca033395deef45aa8fbaee38cb

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewController.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewController.swift:124]: Done in e5177d4705ee7bca033395deef45aa8fbaee38cb

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewModel.swift (hunk 4 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewModel.swift]: Done in 44f51bea278c41365cfae9b31bcc36283f26957d

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17839-44f51be</code>. IPA is available <a href='https://appcenter-filemanagement-distrib1ede6f06e.azureedge.net/964cf11c-05e4-418a-b1d0-700445e8942a/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=v7YRZOjVsiOjdF6cu621JCmVto9qDkP%2BRgo4qjvIrns%3D&se=2022-02-01T15%3A20%3A04Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17839-44f51be</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/423c6e16-2c86-4160-bb33-e08d9cdad3bc/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=emumxNH3n4nrnlwW3ud%2BeD7d0gFafrt7%2BedCROZtQjE%3D&se=2022-02-01T15%3A21%3A57Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Pull request merged**